src/tracer.cpp

```diff
@@ -61,7 +58,12 @@ std::unique_ptr<ot::Span> Tracer::StartSpanWithOptions(ot::string_view operation
   // Get a new span id.
   auto span_id = get_id_();
 
-  auto span_context = SpanContext{span_id, span_id, {}};
+  SpanContext span_context = SpanContext{span_id, span_id, nullptr /* No sampling_priority */, {}};
+  // See the comment in propagation.h on nginx_opentracing_compatibility_hack_.
+  if (operation_name == "dummySpan") {
```

Comment:
Should an issue be opened with nginx-opentracing about this?
It's a fairly wide-reaching hack and eventually devolves to this comparison.
(Also what happens if I use that highly desirable value for operation name?)

Reply:
I might open an issue on this this tracer, and then resolving that will involve commits upstream.

My idea on how to fix it is polymorphism. Have a subclass of ot::Tracer that has a "get propagation headers" method. Opentracing-nginx then still accepts ot::Tracer, but tries a dynamic_cast to see if it can call getPropagationHeaders before falling back to the old hax.

Reply:
https://github.com/DataDog/dd-opentracing-cpp/issues/50